ui/articles-aem-renderer/src/androidTest/java/org/cru/godtools/articles/aem/db/ArticleRoomDatabaseTest.java

```diff
@@ -95,20 +100,70 @@ public void verifyGetAllArticles() {
         assertTrue("No article was saved.", mSavedArticles.size() > 0);
     }
 
-
     @Test
     public void verifyArticleHasAttachment() {
         for (Article article : mSavedArticles) {
             assertTrue(String.format("Article %s has no attachment", article.mTitle),
-                       mAttachmentDao.getTestableAttachmentsByArticle(article.mId).size() > 0);
+                    mAttachmentDao.getTestableAttachmentsByArticle(article.mkey).size() > 0);
         }
     }
 
-
     @Test
     public void verifyManifestHasArticles() {
         assertTrue("Manifest 0 should have articles", mAssociationDao.getTestableArticlesByManifestID("0").size() > 0);
         assertTrue("Manifest 1 should have articles", mAssociationDao.getTestableArticlesByManifestID("1").size() > 0);
         assertTrue("Manifest 2 should have articles", mAssociationDao.getTestableArticlesByManifestID("2").size() > 0);
     }
+
+
+    @Test
+    public void verifyArticleParseLogic() {
+
+        try {
+            Uri url = Uri.parse("https://stage.cru.org/content/experience-fragments/questions_about_god/english.999.json");
+
+            JSONObject jsonObject = new JSONObject(IOUtils.toString(
+                    new URI(url.toString()))
+            );
+
+            HashMap<String, Object> results = new ArticleParser(jsonObject).execute();
+
+            assertTrue(results.containsKey(ArticleParser.ATTACHMENT_LIST_KEY));
+            assertTrue(results.containsKey(ArticleParser.ARTICLE_LIST_KEY));
+
+            List<Article> resultsArticles = (List<Article>) results.get(ArticleParser.ARTICLE_LIST_KEY);
+            assertTrue(resultsArticles.size() > 0);
+            List<Attachment> resultAttachments = (List<Attachment>) results.get(ArticleParser.ATTACHMENT_LIST_KEY);
+            assertTrue(resultAttachments.size() > 0);
+        } catch (MalformedURLException | JSONException | URISyntaxException e) {
+            fail(e.getMessage());
+        } catch (IOException e) {
+            fail(e.getMessage());
+        }
+
+    }
+
+    @Test
+    public void verifyAttachmentsAreSaved() {
+        for (Article article : mSavedArticles) {
+            for (Attachment attachment : mAttachmentDao.getTestableAttachmentsByArticle(article.mkey)) {
+                try {
+                    AEMDownloadManger
```

Comment:
you are testing AEMDownloadManager functionality, so this should be in an `AEMDownloadManagerTest` test class instead of in the `ArticleRoomDatabaseTest` class